test/java/seedu/addressbook/commands/FindByNumberCommandTest.java

```diff
@@ -0,0 +1,50 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+public class FindByNumberCommandTest {
+
+    private final AddressBook addressBook = new TypicalPersons().getTypicalAddressBook();
+    private final TypicalPersons td = new TypicalPersons();
+
+    @Test
+    public void execute() throws IllegalValueException {
+        //same number, same case: matched
+        assertFindByNumberCommandBehavior("91119111", Arrays.asList(td.amy));
+
+        //different number, no match
```

Comment:
Good that you updated the tests but moving forward, do think about other possible test cases for your given command as well.